src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -16476,8 +16514,23 @@ def dt_tz_convert(self, tz: Union[str, tzinfo]) -> None:
         Returns:
             A new QueryCompiler containing values with converted time zone.
         """
-        ErrorMessage.not_implemented(
-            "Snowpark pandas doesn't yet support the method 'Series.dt.tz_convert'"
+
+        def tz_convert_column(
+            column: SnowparkColumn, tz: Union[str, tzinfo]
+        ) -> SnowparkColumn:
+            if tz is None:
+                return convert_timezone(pandas_lit("UTC"), column)
+            else:
+                if isinstance(tz, tzinfo):
+                    tz_name = tz.tzname(None)
+                else:
+                    tz_name = tz
+                return convert_timezone(pandas_lit(tz_name), column)
```

Comment:
Maybe move into `timestamp_utils`? I think this could be used by other functions dealing with timezones.

Reply:
Good suggestion. Done!